### ISSUE TITLE
[BUG] Inconsistent eval metric results 

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/models#144]
I am training a DLRM model with my custom dataset. I get the following metric print outs during training:
```
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
1315/1315 [==============================] - 60s 40ms/step - y/binary_classification_task/precision: 0.0108 - y/binary_classification_task/recall: 0.0038 - y/binary_classification_task/binary_accuracy: 0.9879 - y/binary_classification_task/auc: 0.6896 - loss: 0.0575 - regularization_loss: 0.0000e+00 - total_loss: 0.0575
********************
Start evaluation
329/329 [==============================] - 15s 38ms/step - y/binary_classification_task/precision: 0.0000e+00 - y/binary_classification_task/recall: 0.0000e+00 - y/binary_classification_task/binary_accuracy: 0.9910 - y/binary_classification_task/auc: 0.7320 - loss: 0.0480 - regularization_loss: 0.0000e+00 - total_loss: 0.0480
********************

```

The AUC metric printed out for validation dataset is `0.7320` but the precision and recall are still ` 0.0000e+00`. 

I build the DLRM model as follows

```
train['ypred'] = 0
lr =0.0005

for fold in range(5):
    ind_train = train.fold != fold
    ind_valid = train.fold == fold
    
    TRAIN = train.loc[ind_train].reset_index(drop=True)

    VALID = train.loc[ind_valid].reset_index(drop=True)

    print(TRAIN.shape, VALID.shape)    

    dlrm_body = ml.DLRMBlock(schema,
        embedding_dim=16,
        bottom_block=ml.MLPBlock([64, 16]),
        top_block=ml.MLPBlock([64, 32]),
    )
    model = dlrm_body.connect(ml.BinaryClassificationTask("y"))
    optimizer = tf.keras.optimizers.Adam(lr)
    model.compile(optimizer=optimizer, run_eagerly=False)
    
    train_loader = get_dataloader(nvt.Dataset(TRAIN)) 
    losses = model.fit(train_loader, epochs=1)
    model.save("./tmp/my_model")
    model.reset_metrics()

    # Evaluate
    print('*'*20)
    print("Start evaluation")
    eval_loader = get_dataloader(nvt.Dataset(VALID)) 
    eval_metrics = model.evaluate(eval_loader, return_dict=True)

    train.loc[ind_valid, 'ypred'] = model.predict(eval_loader) 
    print("AUC SCORE:")
    print(roc_auc_score(train.loc[ind_valid, 'y'], train.loc[ind_valid, 'ypred']) )
```
